### DISCUSSION TITLE
how to  use ContentRenderer to render external API data

### DISCUSSION THREAD

**Discussion by leosin** [nuxt/content#1974] [Q&A] [answered]
Can external API data (in markdown format) can be rendered using ContentRenderer with Nuxt Content?

For example: i fetch article data from Directus。and the article is generated using directus markdown editor.

```js
content: "# title \n ## some text....."
```

In the local document (It is the MD document in the content directory), I use the following method, which can work.
```vue
<ContentRenderer :value="content">
  <header>
    <h1>{{ content.title }} <small>{{ content.name }}</small></h1>
    <div class="blocks desc">
      {{ content.description }}
    </div>
  </header>
  <ContentRendererMarkdown :value="content" />
</ContentRenderer>
```
is there any good ideas?

**Comment by Chengyunlai** [comment]
Have you solved it yet? I also want to get the Md data from an external API and render it with nuxt-content.

**Reply by Chengyunlai** [reply]
Wow, it's been about 2 years since I came back to this issue again. Now, I have a resolution (Of course, I don't know if anyone has already raised it 😄 ):
```html
<ContentRenderer :value="apiData" class="prose dark:prose-invert"></ContentRenderer>
```
That `apiData` is external API data,such as the backend API service (golang or java ...) returns a Markdown string

Key Javascript:
```js
const { data: apiData } = await useFetch(`/api/XX`, {
  transform: (res) => {
    
    const parsedMarkdown = marked(res.profile || `
    ## Default Markdown
`);

    return {
      _id: res.XX,
      body: {
        type: 'root',
        children: [
          {
            type: 'element',
            tag: 'div',
            props: {
              innerHTML: parsedMarkdown
            }
          }
        ]
      },
      title: res.XX
    };
  }
});
```
`XX` is real attribute

**Comment by ali4zimi** [comment]
I also needed this, and I used this solution:

https://github.com/nuxt/content/issues/924#issuecomment-1407967034

**Reply by leosin** [reply]
Thank you very much, according to the post you mentioned, the markdown content in the remote api can be displayed.
But there is a problem, the highlighting in markdown, as well as custom modules, cannot be used.

**Reply by ali4zimi** [reply, marked as answer]
For me, it is working. I have created a component named `MarkdownRenderer` with the following content:

```vue
<template>
  <ContentRenderer :value="parsedMarkdown" />
</template>

<script>
import { defineComponent } from 'vue'

import markdownParser from '@nuxt/content/transformers/markdown'

export default defineComponent({
  components: {},

  // define props
  props: {
    md: {
      type: String,
      default: '',
    },
    cid: {
      type: String,
      default: '<some-id>',
    },
  },

  // init
  async setup(props) {
    return {
      parsedMarkdown: await markdownParser.parse(props.cid, props.md),
    }
  },
})
</script>
```

and then call it like this:

```vue
<MarkdownRenderer :md="item.description" />
```

if you are using tailwind, maybe you need to add `prose` class

**Reply by lionel-addvanto** [reply]
Hmm, I still get this error: `[plugin:vite:import-analysis] Failed to resolve import "@nuxt/content/transformers/markdown/"`

**Reply by leosin** [reply]
// @ts-expect-error avoid lint error
import markdownParser from '@nuxt/content/transformers/markdown'

try this : // @ts-expect-error avoid lint error
and use nuxt latest version 3.5.x

**Comment by aatrooox** [comment]
you can use `@nuxtjs/mdc` like this:

after add moudle `@nuxtjs/mdc`

```ts
const content  = "# title \n ## some text....."

<MDC :value="content" tag="section" class="mdc-memo-prose prose"/>
```
now, your custom content render by external api data has same style like your articles
